packages/next/server/hot-reloader.ts

```diff
@@ -236,7 +236,7 @@ export default class HotReloader {
   }
 
   async clean() {
-    return recursiveDelete(join(this.dir, this.config.distDir))
+    return recursiveDelete(join(this.dir, this.config.distDir), /[\\/]cache/)
```

Comment:
Looks like we might need to change the `filter` behavior on `recursiveDelete` for this to work